src/software/pipeline/main_LdrToHdrCalibration.cpp

```diff
@@ -260,6 +284,59 @@ int aliceVision_main(int argc, char** argv)
 
             sampling.analyzeSource(samples, channelQuantization, group_pos);
 
+            std::map<int, std::pair<int, double>> meanLum;
+            std::map<int, std::pair<double, double>> rangeLum;
+            for (int i = 0; i < groupedExposures[group_pos].size(); i++)
+            {
+                meanLum[(int)(groupedExposures[group_pos][i] * 1000)].first = 0;
+                meanLum[(int)(groupedExposures[group_pos][i] * 1000)].second = 0.0;
+                rangeLum[(int)(groupedExposures[group_pos][i] * 1000)].first = 1000.0;
+                rangeLum[(int)(groupedExposures[group_pos][i] * 1000)].second = 0.0;
+            }
+            for (int i = 0; i < samples.size(); i++)
+            {
+                for (int j = 0; j < samples[i].descriptions.size(); j++)
+                {
+                    double lum = 0.2126 * samples[i].descriptions[j].mean[0] + 0.7152 * samples[i].descriptions[j].mean[1] + 0.0722 * samples[i].descriptions[j].mean[2];
+                    meanLum[(int)(samples[i].descriptions[j].exposure * 1000)].second += lum;
+                    meanLum[(int)(samples[i].descriptions[j].exposure * 1000)].first++;
+                    if (lum < rangeLum[(int)(samples[i].descriptions[j].exposure * 1000)].first)
+                    {
+                        rangeLum[(int)(samples[i].descriptions[j].exposure * 1000)].first = lum;
+                    }
+                    if (lum > rangeLum[(int)(samples[i].descriptions[j].exposure * 1000)].second)
+                    {
+                        rangeLum[(int)(samples[i].descriptions[j].exposure * 1000)].second = lum;
+                    }
+                }
+            }
+
+            double minDiffWithLumaTarget = 1000.0;
+            int minDiffWithLumaTargetIdx = 0;
+
+            int k = 0;
+            for (const auto& n : meanLum)
+            {
+                double lumaMean = n.second.second / (double)n.second.first;
+
+                std::cout << '[' << n.first << "] = " << lumaMean << std::endl;
+
+                if (fabs(lumaMean - meanTargetedLuma) < minDiffWithLumaTarget)
+                {
+                    minDiffWithLumaTarget = fabs(lumaMean - meanTargetedLuma);
+                    minDiffWithLumaTargetIdx = k;
+                }
+                ++k;
+            }
+            std::cout << std::endl;
+            for (const auto& n : rangeLum)
+            {
+                std::cout << '[' << n.first << " , " << n.second << "]" << std::endl;
+            }
+            std::cout << std::endl;
+
+            refIndexes.push_back(minDiffWithLumaTargetIdx);
```

Comment:
Maybe better to move all the code used to compute minDiffWithLumaTargetIdx in a dedicated function.